toolkits/spotify/arcade_spotify/tools/utils.py

```diff
@@ -0,0 +1,127 @@
+import httpx
+
+from arcade.core.schema import ToolContext
+from arcade.sdk.errors import ToolExecutionError
+from arcade_spotify.tools.constants import ENDPOINTS, SPOTIFY_BASE_URL
+from arcade_spotify.tools.models import PlaybackState
+
+
+async def send_spotify_request(
+    context: ToolContext,
+    method: str,
+    url: str,
+    params: dict | None = None,
+    json_data: dict | None = None,
+) -> httpx.Response:
+    """
+    Send an asynchronous request to the Spotify API.
+
+    Args:
+        context: The tool context containing the authorization token.
+        method: The HTTP method (GET, POST, PUT, DELETE, etc.).
+        url: The full URL for the API endpoint.
+        params: Query parameters to include in the request.
+        json_data: JSON data to include in the request body.
+
+    Returns:
+        The response object from the API request.
+
+    Raises:
+        ToolExecutionError: If the request fails for any reason.
+    """
+    headers = {"Authorization": f"Bearer {context.authorization.token}"}
+
+    async with httpx.AsyncClient() as client:
+        response = await client.request(method, url, headers=headers, params=params, json=json_data)
+
+    return response
+
+
+def handle_spotify_response(response: httpx.Response, url: str):
```

Comment:
Based on the exception handling Sam put in recently, I think this kind of method is no longer needed at all. Each tool could simply call `response.raise_for_status()` -- see https://www.python-httpx.org/quickstart/#response-status-codes

Reply:
Yep, updated